pkg/modifyresponse_test.go

```diff
@@ -0,0 +1,134 @@
+package transportd
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+)
+
+func TestEnforceRelativeLocation(t *testing.T) {
+	tests := []struct {
+		name         string
+		resp         *http.Response
+		wantErr      bool
+		wantLocation string
+	}{
+		{
+			name: "missing location",
+			resp: &http.Response{
+				Header: http.Header{},
+			},
+			wantErr:      false,
+			wantLocation: "",
+		},
+		{
+			name: "invalid location",
+			resp: &http.Response{
+				Header: http.Header{
+					"Location": []string{"https://[XXX]:notport/path"},
+				},
+			},
+			wantErr:      false,
+			wantLocation: "https://[XXX]:notport/path",
+		},
+		{
+			name: "relative location",
+			resp: &http.Response{
+				Header: http.Header{
+					"Location": []string{"/path/to/api?q=1"},
+				},
+			},
+			wantErr:      false,
+			wantLocation: "/path/to/api?q=1",
+		},
+		{
+			name: "relative location (trailing)",
+			resp: &http.Response{
+				Header: http.Header{
+					"Location": []string{"/path/to/api/?q=1"},
+				},
+			},
+			wantErr:      false,
+			wantLocation: "/path/to/api/?q=1",
+		},
+		{
+			name: "absolute location",
+			resp: &http.Response{
+				Header: http.Header{
+					"Location": []string{"https://localhost/path/to/api?q=1"},
+				},
+			},
+			wantErr:      false,
+			wantLocation: "/path/to/api?q=1",
+		},
+		{
+			name: "relative location (trailing)",
```

Comment:
question: should this test be named absolute?

Reply:
Yeah, I fixed it.